cpp/src/io/text/data_chunk_source_factories.cpp

```diff
@@ -32,7 +33,7 @@ namespace {
 
 struct host_ticket {
   cudaEvent_t event;
-  cudf::detail::pinned_host_vector<char> buffer;
+  std::unique_ptr<cudf::detail::host_vector<char>> buffer;
```

Comment:
Are we using a `unique_ptr` to indicate a state that would be better represented by an `optional`?

Reply:
The buffer can be given as having zero size, which should give the same effect as `nullptr`. Shouldn't it?

Reply:
It would have to use cudf::get_default_stream, so the object is effectively useless. Calling resize on a default-initialized object would lead to default stream use.

Reply:
> Oh, that would be bad if we can't put it into a vector.
We can put them in a vector. Some vector operations would not be possible (e.g. resize) but otherwise it's fine.

Reply:
Changed this with a default ctor for `host_ticket`. No std::optional or std::unique_ptr any more.